src/openms/include/OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/Container.h

```diff
@@ -0,0 +1,249 @@
+    void sortByIntensity();
+
+    /** @brief re-normalizes the sum of the probabilities of the isotopes to 1
+
+            The re-normalisation is needed as in distributions with a lot of isotopes (and with high max isotope)
+            the calculations tend to be inexact.
+    */
+    void renormalize();
+
+    /** @brief Trims the right side of the isotope distribution to isotopes with a significant contribution.
+
+            If the isotope distribution is calculated for large masses (and with high max isotope)
+            it might happen that many entries contain only small numbers. This function can be
+            used to remove these entries.
+
+            Do consider normalising the distribution afterwards.
+    */
+    void trimRight(double cutoff);
+
+    /** @brief Trims the left side of the isotope distribution to isotopes with a significant contribution.
+
+            If the isotope distribution is calculated for large masses (and with high max isotope)
+            it might happen that many entries contain only small numbers. This function can be
+            used to remove these entries.
+
+            Do consider normalising the distribution afterwards.
+    */
+    void trimLeft(double cutoff);
+
+
+    
+    bool isNormalized() const;
+
+
+    bool isConvolutionUnit() const;
+    //@}
+
+    /// @name Operators
+    //@{
+    /// Assignment operator
+    IsotopeDistribution & operator=(const IsotopeDistribution & isotope_distribution);
+
+    /// equality operator, returns true if the @p isotope_distribution is identical to this, false else
+    bool operator==(const IsotopeDistribution & isotope_distribution) const;
+
+    /// inequality operator, returns true if the @p isotope_distribution differs from this, false else
+    bool operator!=(const IsotopeDistribution & isotope_distribution) const;
+    //@}
+
+    /// @name Iterators
+    //@{
+    inline Iterator begin() { return distribution_.begin(); }
+
+    inline Iterator end()   { return distribution_.end(); }
+
+    inline ConstIterator begin() const { return distribution_.begin(); }
+
+    inline ConstIterator end() const { return distribution_.end(); }
+
+    inline ReverseIterator rbegin() { return distribution_.rbegin(); }
+
+    inline ReverseIterator rend()   { return distribution_.rend(); }
+
+    inline ConstReverseIterator rbegin() const { return distribution_.rbegin(); }
+
+    inline ConstReverseIterator rend() const { return distribution_.rend(); }
+
+    inline void insert(const Peak1D::CoordinateType& mass, const Peak1D::IntensityType& intensity)
+    {
+      distribution_.push_back(Peak1D(mass, intensity));
+    }
+    //@}
+
+    /// @name Convolutional Dummy Operators
+    //@{
+    /// operator which adds this distribution and the @p isotope_distribution to return IsotopeDisribution (similar to convolve distributions)
+    IsotopeDistribution operator+(const IsotopeDistribution & isotope_distribution) const;
```

Comment:
I did not check if these are now implemented in a generic way. otherwise move out of base class